src/components/TopBar.js

```diff
@@ -108,29 +93,26 @@ export default _.flow(
   }
 
   hideNav() {
-    this.setState({ navShown: false, userMenuOpen: false })
+    this.setState({ navShown: false, openUserMenu: false, openLibraryMenu: false, openSupportMenu: false })
     document.body.classList.remove('overlayOpen', 'overHeight')
   }
 
   buildNav() {
-    const { authState: { isSignedIn, profile } } = this.props
+    const { authState: { isSignedIn, profile,  profile: { firstName = 'Loading...', lastName = '' }  } } = this.props
     const { trialState } = profile
+    const { openLibraryMenu, openSupportMenu, openUserMenu } = this.state
 
-    const librarySubItem = (linkName, iconName, label) => h(Clickable, {
-      style: styles.nav.subItem,
+    const dropDownSubItem = (linkToPage, title, onClick, props) => h(MenuButton, {
```

Comment:
This abstraction would be better as a component, rather than a function. That would make these arguments into props, and most of them could just be passthroughs (`onClick`, `href`, `children`). Also, since this is overriding almost all the visual styles from `MenuButton`, it would probably be cleaner to just render a `Clickable`here.